lab_3_generate_by_ngrams/main.py

```diff
@@ -143,6 +188,14 @@ def _decode(self, corpus: tuple[int, ...]) -> Optional[tuple[str, ...]]:
         In case of corrupt input arguments, None is returned.
         In case any of methods used return None, None is returned.
         """
+        if not isinstance(corpus, tuple) or not corpus:
+            return None
+        decoded = []
+        for token_id in corpus:
+            decoded.append(self.get_token(token_id))
+        if None in decoded:
```

Comment:
move to the loop. and return as soon as posible?

Reply:
attempted